src/File.cpp

```diff
@@ -124,6 +124,13 @@ void matioCpp::File::close()
 
 bool matioCpp::File::open(const std::string &name, matioCpp::FileMode mode)
 {
+    struct stat info;
+
+    if (stat(name.c_str(), &info) != 0)
```

Comment:
Does this work on Windows? I do not remember if we are using C++17 here, but `std::filesystem` also can provide this info.

Reply:
Yes, it is tested in CI. I already used ``stat`` in the same class to check if a folder existed. I initially developed the library using ``cpp14``, and I was in doubt about using ``filesystem``, but to keep the modifications small I preferred this way.